src/plugins/data_views/server/content_management/data_views_storage.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { ContentStorage, StorageContext } from '@kbn/content-management-plugin/server';
+import type { SearchQuery } from '@kbn/content-management-plugin/common';
+
+import { SavedObjectsFindOptions } from '@kbn/core-saved-objects-api-server';
+import type { DataViewAttributes } from '../../common';
+import type {
+  DataViewGetOut,
+  DataViewCreateIn,
+  DataViewCreateOut,
+  DataViewCreateOptions,
+  DataViewUpdateIn,
+  DataViewUpdateOut,
+  DataViewUpdateOptions,
+  DataViewDeleteOut,
+  DataViewSearchOut,
+  DataViewSearchOptions,
+} from '../../common/content_management';
+import { DataViewSOType } from '../../common/content_management/constants';
+
+const savedObjectClientFromRequest = async (ctx: StorageContext) => {
+  if (!ctx.requestHandlerContext) {
+    throw new Error('Storage context.requestHandlerContext missing.');
+  }
+
+  const { savedObjects } = await ctx.requestHandlerContext.core;
+  return savedObjects.client;
+};
+
+export class DataViewsStorage implements ContentStorage {
```

Comment:
@sebelga This would be my starting point for abstracting ContentStorage for use with Saved Objects. Is there anything in particular that I should keep an eye on regarding the Maps implementation? It currently has versioning code so I'd like to make sure I we're on the same page regarding what the end product should look like.

Reply:
Overall looks good. You would probably need to add all the BWC logic for up/down transform + validate the inputs/outputs. 
You would also need to implement the `bulkGet` method.